staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -91,6 +92,12 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 			KeepAlive: 30 * time.Second,
 		}).DialContext
 	}
+
+	// If we use a cert callback, we need to handle certificate rotation properly
```

Comment:
```
if certFile >0 and keyfile > 0 and !function-override{
    leak a gofunc to watch the file and create the dialer
}
```

This would allow a cache to work and allow us to limit ourselves to a bounded number of leaked gofuncs based on number of files.